assembly/plaindate.ts

```diff
@@ -47,16 +52,26 @@ export class PlainDate {
     if (isString<T>()) {
       // @ts-ignore: cast
       return this.fromString(<string>date);
-    }
-    if (isReference<T>()) {
-      if (date instanceof PlainDate) {
-        return new PlainDate(date.year, date.month, date.day);
+    } else {
+      if (isReference<T>()) {
+        if (date instanceof PlainDate) {
+          return new PlainDate(date.year, date.month, date.day);
+        }
       }
+      throw new TypeError("invalid date type");
     }
-    throw new TypeError("invalid date type");
   }
 
-  constructor(readonly year: i32, readonly month: i32, readonly day: i32) {}
+  constructor(readonly year: i32, readonly month: i32, readonly day: i32) {
+    // if (!(
+    //   checkRange(month, 1, 12) &&
```

Comment:
@ColinEberhardt that's pretty strange. After uncomment this simple check: `if (!checkRange(month, 1, 12)
) throw error`

test `it(`Temporal.PlainDate.from(${date}) is not the same object)`, () => notEqual(PlainDate.from(date), date));`
will fail.